src/modules/launcher/mod.rs

```diff
@@ -292,19 +300,26 @@ impl Module<gtk::Box> for LauncherModule {
                     );
                 } else {
                     send_async!(tx, ModuleUpdateEvent::ClosePopup);
+                    let mut minimize_window = false;
+
+                    let search_items = |app_id| {
+                        lock!(items).get(&app_id).and_then(|item| {
+                            item.windows
+                                .iter()
+                                .find(|(_, win)| !win.open_state.is_focused())
+                                .or_else(|| item.windows.first())
+                                .map(|(_, win)| win.id)
+                        })
+                    };
 
                     let id = match event {
-                        ItemEvent::FocusItem(app_id) => {
-                            lock!(items).get(&app_id).and_then(|item| {
-                                item.windows
-                                    .iter()
-                                    .find(|(_, win)| !win.open_state.is_focused())
-                                    .or_else(|| item.windows.first())
-                                    .map(|(_, win)| win.id)
-                            })
-                        }
+                        ItemEvent::FocusItem(app_id) => search_items(app_id),
```

Comment:
Based on the prev comment, that would allow you to combine the focus/minimize branches here:

```suggestion
                        ItemEvent::FocusItem(app_id) => search_items(app_id),
                        ItemEvent::FocusItem(app_id) | ItemEvent::MinimizeItem => search_items(app_id),
```

Reply:
Makes sense! I wanted them on the same branch but I preoccupied myself hoping for some esoteric solution that just doing it in place never crossed my mind